OnlineShopAPI/IdentityServer/Controllers/UserController.cs

```diff
@@ -85,41 +85,58 @@ public async Task<ActionResult<UserDto>> LogIn([FromBody] UserLogInDto userLogIn
             return Ok(userDto);
         }
 
-        [HttpPost("google-auth")]
-        public async Task<ActionResult<UserDto>> GoogleAuthUser(string token)
+        [HttpPost("google-register")]
+        public async Task<ActionResult<UserDto>> GoogleRegisterUser(string token)
         {
             var dto = _tokenService.GetOAuthDtoFromToken(token);
 
             var user = await _uow.UserRepository.GetUserByUsernameAsync(dto.UserName);
 
             UserDto userDto;
 
-            if (user == null)
+            if (user != null)
             {
-                var newUser = _mapper.Map<AppUser>(dto);
+                return BadRequest("User with this name already exists");
+            }
 
-                var result = await _uow.UserRepository.CreateUserAsync(newUser, "pas$worD123456789426734683275235382");
+            var newUser = _mapper.Map<AppUser>(dto);
 
-                if (!result.Succeeded)
-                {
-                    return BadRequest(result.ToErrorsString());
-                }
+            var result = await _uow.UserRepository.CreateUserAsync(newUser, "pas$worD123456789426734683275235382");
 
-                await _uow.UserRepository.AddToRoleAsync(newUser, "buyer");
+            if (!result.Succeeded)
+            {
+                return BadRequest(result.ToErrorsString());
+            }
 
-                var roles = await _uow.UserRepository.GetUserRoles(newUser);
+            await _uow.UserRepository.AddToRoleAsync(newUser, "buyer");
 
-                userDto = _mapper.Map<UserDto>(newUser);
-                var newToken = _tokenService.CreateToken(newUser, roles);
-                userDto.Token = newToken;
-            }
-            else
+            var roles = await _uow.UserRepository.GetUserRoles(newUser);
+
+            userDto = _mapper.Map<UserDto>(newUser);
+            var newToken = _tokenService.CreateToken(newUser, roles);
+            userDto.Token = newToken;
+
+            return Ok(userDto);
+        }
+
+        [HttpPost("google-login")]
```

Comment:
We can keep it as one endpoint, there is no need to split it to register and log in